app/models/account/entry.rb

```diff
@@ -17,6 +17,7 @@ class Account::Entry < ApplicationRecord
   scope :chronological, -> { order(:date, :created_at) }
   scope :reverse_chronological, -> { order(date: :desc, created_at: :desc) }
   scope :without_transfers, -> { where(marked_as_transfer: false) }
+  scope :without_income_transactions_in_liability_accounts, -> { joins(:account).where(amount: 0..).where.not(accounts: { classification: Account.classifications[:liability] }).distinct }
```

Comment:
I think we're on the right track here, but I wouldn't make a scope for this given how specific it is to the `snapshot_transactions` method.

Also, "Income" would be anything _less than_ 0, so I _think_ we need to flip that condition here